### ISSUE TITLE
test_mrc unit test

### ISSUE THREAD

**Issue by garrettwrong** [ComputationalCryoEM/ASPIRE-Python#326]
A developer reported that they had a spurious failure in `test_mrc.py` that went away on rerun.  This test gets run a lot, and should be deterministic. This is the first report.  I was unable to reproduce so far, but will loop it... 

I am going to add some more diagnostics information when it fails so we can get the test 100%.

Right now I am suspicious of `filecmp` as it is a new tool for python3 that I haven't used before, and it has already been a little quirky for my taste.  I may just hash the files and compare them that way if this is the problem.

I am also suspicious of the floating point arithmetic there since I am bitwise comparing. To fix this should just require ensuring we stay in the set of exactly representable numbers for the test.

Both are simple issues once confirmed.

**Comment by garrettwrong** [comment]
I eventually was able to reproduce.  I had to read the source code of `mrcfile` python package, because there was nothing in the official spec that would have caused this...

It seems `mrcfile` auto populates a label with a time stamp.  Specifically:

```
        time = datetime.now().strftime('%Y-%m-%d %H:%M:%S')
        header.label[0] = '{0:40s}{1:>39s} '.format('Created by mrcfile.py',
                                                    time)
```

I was controlling one `mrc.header.time` stamp already, but not this one (I didn't know about it).  If the two calls occurred on different second this would cause the files to not match.

**Comment by garrettwrong** [comment]
Anyway, I am hopeful this will fix this issue.  Good catch @junchaoxia .

**Comment by janden** [comment]
:+1:

**Comment by garrettwrong** [comment]
Closed by #328
